packages/remark-abbr/lib/index.js

```diff
@@ -1,152 +1,247 @@
-import { visit } from 'unist-util-visit'
-
-export default function plugin (options) {
-  const opts = options || {}
-  const expandFirst = opts.expandFirst
-
-  function locator (value, fromIndex) {
-    return value.indexOf('*[', fromIndex)
+import { SKIP, CONTINUE, visit } from 'unist-util-visit'
+import { abbr, abbrTypes } from 'micromark-extension-abbr'
+
+function splitTextByAbbr (textNode, abbrDefinitions, seenAbbreviations, opts) {
+  const expandFirst = opts.expandFirst || false
+  const uniqueAbbreviationMap = new Map()
+  for (const abbreviation of abbrDefinitions) {
+    uniqueAbbreviationMap.set(abbreviation.identifier, abbreviation)
   }
 
-  function inlineTokenizer (eat, value, silent) {
-    const regex = /[*]\[([^\]]*)\]:\s*(.+)\n*/
-    const keep = regex.exec(value)
+  const uniqueAbbreviations = [...uniqueAbbreviationMap.values()]
+
+  const matches = uniqueAbbreviations
+    .map(
+      (abbr) =>
+        /** @type {const} */ ([abbr, textNode.value.indexOf(abbr.identifier)])
+    )
+    .filter(([_abbr, index]) => index >= 0)
+    .map(([abbr, index]) => {
+      const start = index
+      const end = index + abbr.identifier.length - 1
+      return {
+        abbr,
+        start,
+        end,
+        prevChar: textNode.value[start - 1],
+        nextChar: textNode.value[end + 1]
+      }
+    })
+    .filter((match) =>
+      // We don't want to match "HTML" inside strings like "HHHHTMLLLLLL", so check that the
```

Comment:
Looks like this actually existed before, since currently HTML works, but I have no idea how it works. Will have to check, because abbr appears in tree.